graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/builtins/NumberFunctionBuiltins.java

```diff
@@ -157,6 +162,12 @@ protected static boolean isInteger(int arg) {
             return true;
         }
 
+        @SuppressWarnings("unused")
+        @Specialization
+        protected static boolean isInteger(long arg) {
+            return JSRuntime.isIntegralNumber((double) arg);
```

Comment:
I suspect we can just return `true` in this case because a `long` will never be represented as a non-integral `double`, but I can't find a proof for this